test/controller.go

```diff
@@ -241,6 +245,13 @@ func SeedTestData(t *testing.T, ctx context.Context, d Data) (Clients, Informers
 			t.Fatal(err)
 		}
 	}
+	c.Kube.PrependReactor("*", "serviceaccounts", AddToInformer(t, i.ServiceAccount.Informer().GetIndexer()))
```

Comment:
This is similar to the other resources we have pipeline, pipelinerun, etc. and I am guessing this would be needed for all service accounts that a task/pipeline is expecting including `default` 🤔

Reply:
As of today, we don't have the reactor, so the informer are/might be out-of-sync. This can cause errors during tests where the serviceaccount is not found. Having this fixes that problem.

Reply:
but indeed, we should update the unit tests that manually create the ServiceAccounts to use this instead